crates/lib3h/src/transport/websocket/mod.rs

```diff
@@ -17,6 +16,39 @@ use std::{
 };
 
 use url::Url;
+use lib3h_ghost_actor::create_ghost_channel;
+
+/// a connection identifier
+pub type ConnectionId = String;
+pub type ConnectionIdRef = str;
+
+/// Commands that can be sent to an implementor of the Transport trait and handled during `process()`
+#[derive(Debug, PartialEq, Clone)]
+pub enum TransportCommand {
+    Connect(Url, String),
+    Send(Vec<ConnectionId>, Vec<u8>),
+    SendAll(Vec<u8>),
+    Close(ConnectionId),
+    CloseAll,
+    Bind(Url),
+}
```

Comment:
I've pulled these two enums back in again to get the old websocket code working, but I believe this one (`TransportCommand`) can go again easily.